data/transactions/transaction.go

```diff
@@ -570,6 +570,9 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 		if !proto.Heartbeat {
 			return fmt.Errorf("heartbeat transaction not supported")
 		}
+		if tx.HbKeyDilution == 0 {
+			return errors.New("tx.HbKeyDilution is zero")
+		}
```

Comment:
idk, maybe it makes sense to reject HB with empty HbSeed/HbVoteID earlier right here?

Reply:
Agreed, I'm adding empty checks.

Reply:
Reminder for myself; will look for most of the "is this is a free heartbeat, go draconian" checks (the must be empties) to move here.

Reply:
Check now.

Reply:
Should I use the `tx.<FieldName>` construction universally? It's common in WellFormed.

Reply:
I'd be fine with that.